conans/test/integration/cmake_targets_test.py

```diff
@@ -132,3 +132,102 @@ def package_info(self):
         client.run("install . -g cmake")
         bili = load(os.path.join(client.current_folder, "conanbuildinfo.cmake"))
         self.assertIn("-framework Foundation", bili)
+
+    @unittest.skipUnless(platform.system() == "Darwin", "Requires Macos")
+    def custom_apple_framework_test(self):
+        """Build a custom apple framework and reuse it"""
+        client = TestClient()
+        lib_c = r"""
+#include <stdio.h>
+#include "mylib.h"
+
+void hello(){
+    printf("HELLO FRAMEWORK!");
+}
+"""
+
+        lib_h = r"""
+void hello();
+"""
+        conanfile = '''
+import os
+from shutil import copyfile
+from conans import ConanFile, CMake
+
+class MyFrameworkConan(ConanFile):
+
+    options = {"shared": [True, False]}
+    default_options = "shared=True"
+    exports = "*"
+    generators = "cmake"
+    
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+    
+    def package(self):
+        self.copy(pattern="*.h", dst="MyFramework.framework/Headers", keep_path=False)
+        copyfile("lib/libmylib.dylib", os.path.join(self.package_folder, "MyFramework.framework", "MyFramework"))
+    
+    def package_info(self):
+        flag_f_location = '-F "%s"' % self.package_folder
+        self.cpp_info.cflags.append(flag_f_location)
+        self.cpp_info.sharedlinkflags.extend([flag_f_location, "-framework MyFramework"])
+        self.cpp_info.exelinkflags = self.cpp_info.sharedlinkflags
+'''
+
+        cmake = """
+set(CMAKE_CXX_COMPILER_WORKS 1)
+set(CMAKE_CXX_ABI_COMPILED 1)
+project(MyHello C)
+cmake_minimum_required(VERSION 2.8.12)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(KEEP_RPATHS)
+add_library(mylib mylib.c)
```

Comment:
may be use CMake's FRAMEWORK here? https://cmake.org/cmake/help/v3.5/prop_tgt/FRAMEWORK.html 
and also copy full directory above instead of hand-crafting the framework